### PR TITLE
feat(core): cleanup provider interface + add integration docs

### PR DESCRIPTION
**Pull request by AvitalTamir**
Remove these guys:
```
	GetDiscoveryClient() (discovery.DiscoveryInterface, error)
	GetClientset() (kubernetes.Interface, error)
	GetDynamicClient() (dynamic.Interface, error)
	GetGVRCache() (map[string]schema.GroupVersionResource, error)
```

The provider can now be initialized with an optional client set.

Also adds integration docs.

### CODE REVIEW

File: operator/internal/controller/dynamicoperator_controller.go [hunk 1]
```diff
@@ -150,14 +150,45 @@ func (r *DynamicOperatorReconciler) logActiveWatchers() {
 	}
 }
 
+// Add this function to initialize kubernetes clients
```
Comment (naorpeled) [inline operator/internal/controller/dynamicoperator_controller.go:153]:
```suggestion
```

File: operator/internal/controller/dynamicoperator_controller.go [hunk 2]
```diff
@@ -150,14 +150,45 @@ func (r *DynamicOperatorReconciler) logActiveWatchers() {
...
+
+	dynamicClient, err := dynamic.NewForConfig(restConfig)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to create dynamic client: %v", err)
+	}
+
+	return clientset, dynamicClient, nil
+}
+
+// Modify the SetupWithManager function to use the new initialization
```
Comment (naorpeled) [inline operator/internal/controller/dynamicoperator_controller.go:173]:
```suggestion
```

File: pkg/provider/apiserver/provider.go [hunk 3]
```diff
@@ -52,40 +58,58 @@ type apiResponse struct {
...
-	// First try in-cluster config
-	config, err := rest.InClusterConfig()
-	if err != nil {
-		// Fall back to kubeconfig
-		loadingRules := clientcmd.NewDefaultClientConfigLoadingRules()
-		configOverrides := &clientcmd.ConfigOverrides{}
-		kubeConfig := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(loadingRules, configOverrides)
-		config, err = kubeConfig.ClientConfig()
+	// Detect if we're in a test environment
+	isTest := strings.HasSuffix(os.Args[0], ".test") ||
```
Comment (naorpeled) [inline pkg/provider/apiserver/provider.go:62]: wdyt about changing to env vars?
this might be a bit more fragile

**Review by naorpeled — APPROVED** [review]
LGTM,
added some non-blocking comments

**Comment by AvitalTamir** [comment]
@naorpeled thanks for the review! I think there is one more we can get rid of but I'm failing some tests, will update

**Pull request merged**